backend/src/announcement/announcement.service.ts

```diff
@@ -36,42 +41,77 @@ export class AnnouncementServices {
     }
   }
   public async createAnnouncement(
+    user: User,
     data: AnnouncementDTO
   ): Promise<AnnouncementDTO> {
     try {
       const announcement = await this.prisma.announcement.create({
         data,
       });
+      this.eventEmitter.emit('create.logs', {
+        entityName: 'announcement',
+        action: ActionENUM.ADDED,
+        username: user.name,
+        priority: PriorityENUM.IMPORTANT,
+      });
+
       return AnnouncementServices.convertToDTO(announcement);
     } catch (error) {
       throw new BadRequestException();
     }
   }
   public async updateAnnouncement(
+    user: User,
     id: string,
     data: AnnouncementDTO
   ): Promise<AnnouncementDTO> {
     try {
+      // DONT KNOW IF THIS IS THE BEST WAY TO DO IT
+      const oldValue = await this.prisma.announcement.findUnique({
```

Comment:
I'm doubtful if this is an ideal approach in saving the `oldValue` of an entity before updating it. I would truly much appreciate it if yall can give me some suggestions on how to deal with this matter (if ekis jud ning ako gibuhat) XD

Reply:
i think it's fine for now. im not really sure what would be the better solution but we'll see 🤔